app/controllers/companies_controller.rb

```diff
@@ -45,10 +61,72 @@ def destroy
     redirect_to companies_path
   end
 
-
   private
 
   def company_params
     params.require(:company).permit(:name, :city)
   end
+
+  def header_index
```

Comment:
same with these methods - i think you've extracted a little too much into these methods. can we simplify this? even if it's at the expense of some repetitive code?